### ISSUE TITLE
[AIR] [Data] Guide to move from DataLoader to Ray Data

### ISSUE THREAD

**Issue by amogkam** [ray-project/ray#31418]
### Description

We have a guide to move existing Pytorch training code to Ray AIR TorchTrainer (https://docs.ray.io/en/latest/ray-air/examples/convert_existing_pytorch_code_to_ray_air.html).

However one aspect that is missing is specifically the data ingest part- how to move from PyTorch DataLoader to Ray Datasets (and what value does Ray Datasets provide?). Specifically for CV workloads.

Generally, pytorch data ingest looks like the following:
1. A list of S3 paths to images/labels. Or a parquet file with the following columns: S3 paths to images, label, additional metadata
2. Inside the training loop, shuffle the datasets for each epoch. Optionally sample the dataset based on the metadata.
3. Inside the training loop, for each batch, do the following: Load the images from S3. Preprocess the images. Apply random per-epoch augmentations.

We should show how to do the above setup with AIR ingest APIs and map from torch dataloader concepts to Ray AIR/Data concepts:
1. Read the parquet file
2. Use stream api and global shuffling in AIR DatasetConfig.
3. Specify a global preprocessor
4. Specify per-epoch preprocessor.

Specifically, the value adds that Ray Data provides:
1. Overlapping CPU preprocessing with GPU training.
2. Scale out CPU preprocessing independently from GPU training (using more CPU only nodes for preprocessing).

**Comment by divyanshk** [comment]
Hi. Stumbled upon this. Curious if someone can point me to more pointers for me to learn the value add mentioned above ? Curious to learn how RayData overlaps preprocessing, and provides GPU independent scaling. Thanks.
